library/src/com/google/maps/android/kml/KmlFeatureParser.java

```diff
@@ -0,0 +1,347 @@
+            eventType = mParser.next();
+        }
+        return new KmlPoint(coordinate);
+    }
+
+    /**
+     * Creates a new KmlLineString object
+     *
+     * @return KmlLineString object
+     */
+    private KmlLineString createLineString() throws XmlPullParserException, IOException {
+        ArrayList<LatLng> coordinates = new ArrayList<LatLng>();
+        int eventType = mParser.getEventType();
+        while (!(eventType == END_TAG && mParser.getName().equals("LineString"))) {
+            if (eventType == START_TAG && mParser.getName().equals("coordinates")) {
+                coordinates = convertToLatLngArray(mParser.nextText());
+            }
+            eventType = mParser.next();
+        }
+        return new KmlLineString(coordinates);
+    }
+
+    /**
+     * Creates a new KmlPolygon object. Parses only one outer boundary and no or many inner
+     * boundaries containing the coordinates.
+     *
+     * @return KmlPolygon object
+     */
+    private KmlPolygon createPolygon() throws XmlPullParserException, IOException {
+        // Indicates if an outer boundary needs to be defined
+        Boolean isOuterBoundary = false;
+        ArrayList<LatLng> outerBoundary = new ArrayList<LatLng>();
+        ArrayList<ArrayList<LatLng>> innerBoundaries = new ArrayList<ArrayList<LatLng>>();
+        int eventType = mParser.getEventType();
+        while (!(eventType == END_TAG && mParser.getName().equals("Polygon"))) {
+            if (eventType == START_TAG) {
+                if (mParser.getName().matches(BOUNDARY_REGEX)) {
+                    isOuterBoundary = mParser.getName().equals("outerBoundaryIs");
+                } else if (mParser.getName().equals("coordinates")) {
+                    if (isOuterBoundary) {
+                        outerBoundary = convertToLatLngArray(mParser.nextText());
+                    } else {
+                        innerBoundaries.add(convertToLatLngArray(mParser.nextText()));
+                    }
+                }
+            }
+            eventType = mParser.next();
+        }
+        return new KmlPolygon(outerBoundary, innerBoundaries);
+    }
+
+    /**
+     * Creates a new KmlMultiGeometry object
+     *
+     * @return KmlMultiGeometry object
+     */
+    private KmlMultiGeometry createMultiGeometry() throws XmlPullParserException, IOException {
+        ArrayList<KmlGeometry> geometries = new ArrayList<KmlGeometry>();
+        // Get next otherwise have an infinite loop
+        int eventType = mParser.next();
+        while (!(eventType == END_TAG && mParser.getName().equals("MultiGeometry"))) {
+            if (eventType == START_TAG && mParser.getName().matches(GEOMETRY_REGEX)) {
+                geometries.add(createGeometry(mParser.getName()));
+            }
+            eventType = mParser.next();
+        }
+        return new KmlMultiGeometry(geometries);
+    }
+
+    /**
+     * Creates a bound for a LatLonBox and sets the GroundOverlay to contain this bound
```

Comment:
this only sets the rotation on the groundoverlay though?  I get why you've done (because you can only return one result) but it's a bit weird that this sets the rotation but returns the latlngbounds.